Source/IGListAdapter.m

```diff
@@ -516,7 +519,18 @@ - (void)updateBackgroundViewWithItemCount:(NSUInteger)itemCount {
         [_collectionView.backgroundView removeFromSuperview];
         _collectionView.backgroundView = backgroundView;
     }
-    _collectionView.backgroundView.hidden = itemCount > 0;
+    _collectionView.backgroundView.hidden = shouldHide;
+}
+
+- (BOOL)itemCountIsZero {
+    __block BOOL isZero = YES;
+    [self.sectionMap enumerateUsingBlock:^(id  _Nonnull object, IGListSectionController<IGListSectionType> * _Nonnull sectionController, NSInteger section, BOOL * _Nonnull stop) {
```

Comment:
hmm, can we? it seems like `id` is the best we can do based on the signature of `[IGListSectionMap enumerateUsingBlock:]`, but quite possibly I am missing something...

```
- (void)enumerateUsingBlock:(void (^)(id object, IGListSectionController<IGListSectionType> *sectionController, NSInteger section, BOOL *stop))block;
```

https://github.com/Instagram/IGListKit/blob/master/Source/Internal/IGListSectionMap.h#L103